### DISCUSSION TITLE
How to use the Upload Certificate​ with aab

### DISCUSSION THREAD

**Discussion by megajogos** [airsdk/Adobe-Runtime-Support#1553] [Q&A]
Any one knows how to convert / use upload_cert.der for signing AAB?

Would be nice to have this documentation here: https://airsdk.dev/docs/tutorials/platform/android/packaging-android-app-bundles#upload-certificate

**Comment by marchbold** [comment]
Are you talking about the one you download from the play console? You can't really use that as it's just the public key and you'll need the private key to sign the app. 

Here are some notes I've put together for the upload key and posted on the starling forum a few days ago.

There are two certificates/keys with the new process:

- one is used by Google to sign your application when delivering to users (this should be your current certificate you used to sign apk's in the past);
- the other is the "upload key" that is used to sign your aab to verify the package when uploading to google play;

These certificates can be the same! And if you follow the defaults when migrating they will be, i.e. Google will use the same certificate for the upload certificate and the app signing certificate. You can check this by comparing the fingerprints in the app integrity section of the play console.

In order to use a different upload certificate you can change the "upload key" when migrating to Play App Signing by selecting the "create a new upload key (optional)" step and following the additional instructions. 

If you have already been through the process and are using Play App Signing, then you'll need to "request key upgrade" to your signing key. During this process you can create a new upload certificate (just using normal create a self signed certificate process) and send this to Google. 

This is generally not needed however we had to do it with one of our clients who still had a 1024 bit certificate. We created a new "upload key" as a 2048 certificate to use when building with AIR and then let Google handle the 1024 bit certificate. This removed the need for us to use java 1.8.112 for the old certificate.
